### DISCUSSION TITLE
Remaster versions of the same album should be findable in the MA library

### DISCUSSION THREAD

**Discussion by busywait** [music-assistant/support#2180] [Feature requests and ideas]
In MA 2.0.0b143 and earlier if two or more Original and "Remaster" release versions of the same album are found then only one will be visible in the library because they are all considered the same "Recording" by MusicBrainz, 

The tracks are deduplicated so only one item for each track is listed in the library.

This happens because MA follows the MusicBrainz schema in organising albums into "Recordings" and considers remastered versions to be the same recording. 
> separate recordings should not be created for remastered tracks, since remastered tracks generally feature the original recording with different mastering applied. Remastering should be described using the [remaster relationship type](https://musicbrainz.org/relationship/48e327b5-2d04-4518-93f1-fed5f0f0fa3c) between releases, or in the release annotation where tracks are mastered differently across a release.
[MusicBrainz Release Documentation](https://musicbrainz.org/doc/Style/Recording)

I would like to be able to at least find earlier Releases of a Recording in the MA library, and to be sure of which track versions are playing when I select a Release in the MA library.

Additionally, although "Remix" tracks are preserved as distinct recordings in MA, they become interleaved with the tracks from the original recording under a single Album item in the MA library.

An example: band Marillion, album Clutching at Straws:
1/ Clutching at Straws, released 1987
2/ Clutching at Straws (1999 24-bit remaster), released 2001
3/ Clutching at Straws (2018 remix), released 2019

The tracks from either 1 or 2 are not visible in my library, and all tracks (including from 3) are listed under a single Album entry "Clutching at Straws"
![image](https://github.com/music-assistant/hass-music-assistant/assets/3671165/b9332d47-7e9e-4101-9811-d72d119bf969)

(Strange behaviour for Track 1 in this screenshot - all three track versions are listed just for Track 1)

I've verified the behaviour with tracks tagged using Picard, and added all MB IDs and metadata that is offered for my files.

**Comment by dragos250** [comment]
I have the same issue where I want to play different releases of the same album (CD rip or Vinyl rip, for example). They are all comingled.

**Comment by marcelveldt** [comment]
The simple answer to this is to delete the MusicBrainz Release and/or Recording ID's from these, because that is what makes it matched.
If you put your own metadata in the tags such as : "My Great Album (1960 vinyl version)" Music Assistant will always prefer that.

Its now just matching on the strong musicbrainz identifier, possibly done by an automatic tagger like Picard.
Its kind of strange btw that MusicBrainz sees a remaster of a recording as the same recording. Maybe there should be something like RedordingMaster or whatever in between

**Reply by marcelveldt** [reply]
As a follow-up: In case you want to override MusicBrainz db:

- Remove MusicBrainz Release ID and Recording ID
- Keep MusicBrainz Artist ID
- Remove ISRC (as that is also used as strong identifier for tracks)
- Remove barcode (as that is also used as string identifier for albums)
- Because there is no version specific ID3 tag, place the version between brackets in the title

**Reply by OzGav** [reply]
This info will be in the next version of the docs

**Comment by bluefuzz** [comment]
> * Remove MusicBrainz Release ID and Recording ID
> * Keep MusicBrainz Artist ID
> * Remove ISRC (as that is also used as strong identifier for tracks)
> * Remove barcode (as that is also used as string identifier for albums) 
> * Because there is no version specific ID3 tag, place the version between brackets in the title

This is only possible for local library content. And even then it is unfeasible for people with large libraries to micro-manage every piece of metadata pertaining to MusicBrainz or other tags. What about the metadata of tracks from the streaming services? That can't be edited ...

It would be more useful to be able to give MA some rules as to how it should select the 'best' track based on user-selectable criteria. And any automated selection needs to able to be overridden on a case by case basis – i.e. play this album from Spotify regardless if MA thinks there is a 'better' version on Tidal or in my local library. The metadata from both the streaming services and certainly MusicBrainz simply contains far too many errors (even if you accept MB's tag guidelines) to be reliable. Even though I am quite careful tagging my local library content I'm pretty sure it too is by no means perfect.

**Reply by marcelveldt** [reply]
We already handle this perfectly fine - it has been a 5 years quest to deal with metadata from stream providers etc.
This specific request/topic is about remasters of the same recording if musicbrainz tags are present in local files. In fact, streaming providers handle this perfectly fine. Remasters are just different versions, as it should be.

**Comment by andrewaylett** [comment]
Distinct releases _should_ have distinct barcodes, but I'm certain that there are examples where they won't.  In general, I'd probably prefer/suggest only using the release ID as the strong identifier for an album and only using track ID as the strong identifier for a track.

Recording ID, ISRC, (or even work ID) may be useful for avoiding playing the same thing twice in a row when shuffling or in a radio mode, but we know they're insufficient for distinguishing distinct audio so we should avoid using them as strong identifiers for deduplicating files.

**Reply by marcelveldt** [reply]
This is already how it works - problem is that MusicBrainz identifies a remaster of the SAME recording as the same recording while that is debatable
